extensions/wikidata/src/org/openrefine/wikidata/schema/WbEntityVariable.java

```diff
@@ -0,0 +1,95 @@
+package org.openrefine.wikidata.schema;
+
+import org.openrefine.wikidata.qa.QAWarning;
+import org.openrefine.wikidata.schema.entityvalues.ReconEntityIdValue;
+import org.openrefine.wikidata.schema.entityvalues.ReconItemIdValue;
+import org.openrefine.wikidata.schema.entityvalues.ReconMediaInfoIdValue;
+import org.openrefine.wikidata.schema.entityvalues.ReconPropertyIdValue;
+import org.openrefine.wikidata.schema.exceptions.SkipSchemaExpressionException;
+import org.wikidata.wdtk.datamodel.helpers.Datamodel;
+import org.wikidata.wdtk.datamodel.implementation.EntityIdValueImpl;
+import org.wikidata.wdtk.datamodel.implementation.PropertyIdValueImpl;
+import org.wikidata.wdtk.datamodel.implementation.*;
+import org.wikidata.wdtk.datamodel.interfaces.EntityIdValue;
+import org.wikidata.wdtk.datamodel.interfaces.ItemIdValue;
+import org.wikidata.wdtk.datamodel.interfaces.MediaInfoIdValue;
+import org.wikidata.wdtk.datamodel.interfaces.PropertyIdValue;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.google.refine.model.Cell;
+import com.google.refine.model.Recon.Judgment;
+
+/**
+ * An Entity that depends on a reconciled value in a column.
+ *
+ */
+
+public class WbEntityVariable extends WbVariableExpr<EntityIdValue> {
+
+    @JsonCreator
+    public WbEntityVariable() {
+
+    }
+
+	/**
+     * Constructs a variable and sets the column it is bound to. Mostly used as a
+     * convenience method for testing.
+     *
+     * @param columnName
+     *            the name of the column the expression should draw its value from
+     */
+    public WbEntityVariable(String columnName) {
+        setColumnName(columnName);
+    }
+
+    @Override
+    public EntityIdValue fromCell(Cell cell, ExpressionContext ctxt)
+            throws SkipSchemaExpressionException {
+        if (cell.recon != null
+                && (Judgment.Matched.equals(cell.recon.judgment) || Judgment.New.equals(cell.recon.judgment))) {
+            if (cell.recon.identifierSpace == null || !cell.recon.identifierSpace.equals(ctxt.getBaseIRI())) {
+                QAWarning warning = new QAWarning("invalid-identifier-space", null, QAWarning.Severity.INFO, 1);
+                warning.setProperty("example_cell", cell.value.toString());
+                ctxt.addWarning(warning);
+                throw new SkipSchemaExpressionException();
+            }
+            EntityIdValue entityIdValue = EntityIdValueImpl.fromId(cell.getReconIdString(), cell.recon.identifierSpace);
+            if (entityIdValue instanceof ItemIdValue) {
+                return new ReconItemIdValue(cell.recon, cell.value.toString());
+            } else if (entityIdValue instanceof MediaInfoIdValue) {
+                return new ReconMediaInfoIdValue(cell.recon, cell.value.toString());
+            } else if (entityIdValue instanceof PropertyIdValue) {
+                return new ReconPropertyIdValue(cell.recon, cell.value.toString());
+            }
```

Comment:
Nice! I think this should work for cells that are `Matched`, but not the ones that are set to `New`.
For the `New` case I do not think there is a satisfactory way to implement this yet, because there is nothing in the `recon` object that can tell us which type of entity we are supposed to create. So I would propose that for now, cells marked as `New` are assumed to be about items (so that the current functionality is preserved), and we should be able to fix this later on when we do #4270.